src/v/archival/ntp_archiver_service.cc

```diff
@@ -81,6 +85,14 @@ ntp_archiver::ntp_archiver(
     if (_partition && _partition->is_read_replica_mode_enabled()) {
         _bucket = _partition->get_read_replica_bucket();
     }
+
+    // Only run the housekeeping task for non-read replicas
+    if (_partition && !_partition->is_read_replica_mode_enabled()) {
```

Comment:
Maybe this should be in run_upload_loop:
- Usually things like spawning background work isn't done in constructors (prefer ::start methods or in this case the run loop method)
- If it's spawned on the same path as the rest of the upload machinery, it can share the upgrade handling in https://github.com/redpanda-data/redpanda/pull/6913

Reply:
Yep, housekeeping moved into the upload loop now.